frontend/js/tests/cb-review/CBReviewModal.test.tsx

```diff
@@ -69,314 +54,269 @@ const apiRefreshSpy = jest
   .spyOn(testAPIService, "refreshAccessToken")
   .mockResolvedValue("this is new token");
 
-const props: CBReviewModalProps = {
-  listen,
-};
-
-const niceModalProps: NiceModalHocProps = {
-  id: "fnord",
-  defaultVisible: true,
-};
-
 describe("CBReviewModal", () => {
+  const user = userEvent.setup();
+  userEvent.setup();
+
   afterEach(() => {
     jest.clearAllMocks();
   });
+
   it("renders the modal correctly", async () => {
-    const wrapper = mount(
-      <GlobalAppContext.Provider value={globalProps}>
-        <NiceModal.Provider>
-          <CBReviewModal {...niceModalProps} {...props} />
-        </NiceModal.Provider>
-      </GlobalAppContext.Provider>
+    render(<NiceModal.Provider />);
+    await act(async () => {
+      NiceModal.show(CBReviewModal, { ...props });
```

Comment:
Do we need `await` here?

Reply:
Yes indeed. I'm still getting used to how to properly deal with asynchronous components and API calls, and running into some issues.
However act does not return a Promise so that would indeed be a big no-no !